lib/iris/coords.py

```diff
@@ -1373,33 +1446,26 @@ def from_regular(cls, zeroth, step, count, standard_name=None,
             bounds values will be defined. Defaults to False.
 
         """
-        coord = DimCoord.__new__(cls)
-
-        coord.standard_name = standard_name
-        coord.long_name = long_name
-        coord.var_name = var_name
-        coord.units = units
-        coord.attributes = attributes
-        coord.coord_system = coord_system
-        coord.circular = circular
-
-        points = (zeroth+step) + step*np.arange(count, dtype=np.float32)
+        points = (zeroth + step) + step * np.arange(count, dtype=np.float32)
+        # Spoof `iris.util.is_regular`, which only works on coords (which we
+        # don't yet have).
+        diffs = np.diff(points)
+        if not np.allclose(diffs, np.mean(diffs), rtol=0.001):
+            points = (zeroth + step) + step * np.arange(count,
+                                                        dtype=np.float64)
```

Comment:
> But what's the point?!

To avoid the monotonicity checks, which I assume was the original intent of whoever wrote this method. Although maybe they just didn't want to write a method like `is_regular` which worked for points.

> You're still duplicating the functionality of the `__init__`.

No I'm not. The `__init__` method would call that method, as I said.

I think the underlying problem here is a combination of two factors:
1) Python doesn't allow multiple `__init__` methods
2) Users can construct DimCoords directly.

Because of this, the `__init__` method _has_ to do the monotonicity checks, so if you want to avoid them you need a separate method.